Update apply page for style, concision, and clarity (#31197)
hashicorp/terraform: website/docs/cli/commands/apply.mdx

# Command: apply

The `terraform apply` command executes the actions proposed in a Terraform
plan.

> > **Hands On:** Try the [Apply Terraform Configuration](https://learn.hashicorp.com/tutorials/terraform/apply?utm_source=WEBSITE&utm_medium=WEB_IO&utm_offer=ARTICLE_PAGE&utm_content=DOCS) tutorial to for details about how Terraform applies a configuration, how Terraform recovers from errors during apply, and common ways to use this command.

Reviewer: "to for"

```suggestion
> **Hands On:** Try the [Apply Terraform Configuration](https://learn.hashicorp.com/tutorials/terraform/apply?utm_source=WEBSITE&utm_medium=WEB_IO&utm_offer=ARTICLE_PAGE&utm_content=DOCS) tutorial to learn how Terraform applies a configuration, how Terraform recovers from errors during apply, and common ways to use this command.
```

## Usage

Usage: `terraform apply [options] [plan file]`

### Automatic Plan Mode

> When you run `terraform apply` without arguments, Terraform automatically creates a new execution plan as if you had run [`terraform plan`](/cli/commands/plan), prompts you to approve that plan, and takes the indicated actions. You can use all of the [planning modes](/cli/commands/plan#planning-modes) and

Reviewer: Tricky here — it also runs in automatic plan mode if you pass it any number of options, like the plan modes/options we immediately mention. From a CLI taxonomy perspective you could argue flags/options aren't "arguments", but it might be best to avoid the ambiguity. 

```suggestion
When you run `terraform apply` without passing a saved plan file, Terraform automatically creates a new execution plan as if you had run [`terraform plan`](/cli/commands/plan), prompts you to approve that plan, and takes the indicated actions. You can use all of the [planning modes](/cli/commands/plan#planning-modes) and
```

[planning options](/cli/commands/plan#planning-options) to customize how Terraform will create the plan.

You can pass the `-auto-approve` option to instruct Terraform to apply the plan without asking for confirmation.

> !> **Warning:** If you use `-auto-approve`, we recommend making sure that no one can change your infrastructure outside of your current workflow. This minimizes the risk of unpredictable changes and configuration drift.

Reviewer: ```suggestion
!> **Warning:** If you use `-auto-approve`, we recommend making sure that no one can change your infrastructure outside of your Terraform workflow. This minimizes the risk of unpredictable changes and configuration drift.
```

### Saved Plan Mode

When you pass a [saved plan file](/cli/commands/plan#out-filename) to `terraform apply`, Terraform takes the actions in the saved plan without prompting you for confirmation. You may want to use this two-step workflow when [running Terraform in automation](https://learn.hashicorp.com/tutorials/terraform/automate-terraform?in=terraform/automation&utm_source=WEBSITE&utm_medium=WEB_IO&utm_offer=ARTICLE_PAGE&utm_content=DOCS).

Use [`terraform show`](/cli/commands/show) to inspect a saved plan file before applying it.

When using a saved plan, you cannot specify any additional planning modes or options. These options only affect Terraform's decisions about which
actions to take, and the plan file contains the final results of those
decisions.

### Plan Options

Without a saved plan file, `terraform apply` supports all planning modes and planning options available for `terraform plan`.

> -  **[Planning Modes](/cli/commands/plan#planning-modes):** These include `-destroy`, which creates a plan to destroy all remote objects, and `-refresh-only` which creates a plan to update Terraform state and root module output values.

Reviewer: ```suggestion
-  **[Planning Modes](/cli/commands/plan#planning-modes):** These include `-destroy`, which creates a plan to destroy all remote objects, and `-refresh-only`, which creates a plan to update Terraform state and root module output values.
```

Reviewer: By the way, I like the additional context for what these things are! 👍🏼

Author: thank you :)
